buildSrc/src/main/groovy/checkstyle-conventions.gradle

```diff
@@ -0,0 +1,34 @@
+plugins {
+    id("checkstyle")
+}
+project.apply from: "${rootDir}/dependencies.gradle"
+
+configurations {
+    checkstyle
+}
+
+dependencies {
+    checkstyle libraries["checkstyle"]
+    checkstyle libraries["checkstyleSevntu"]
+}
+
+checkstyle {
+    // Use the latest checkstyle version instead of default one
+    def module = project.dependencies.create(libraries["checkstyle"])
+    def version = module.version
+    toolVersion = version
+    configFile = file("${rootDir}/config/checkstyle/checkstyle.xml")
+    maxWarnings = 0
+    maxErrors = 0
+}
+
+checkstyleMain.exclude("**/i18n/Messages*")
+
+// Create a single checkstyle task to make it easier to run both the checkstyleMain, checkstyleTest,
+// and checkstyleSPec targets
+tasks.register("checkstyle") {
+    dependsOn tasks.checkstyleMain
+    dependsOn tasks.checkstyleTest
```

Comment:
It kinda does. Moving checkstyle into a convention plugin changed how gradle executes our tasks, leveraging it's task hierarchy.

Previously running ./gradlew checkstyle executed three tasks. checkstyleMain, checkstyleTest and checkstyleSpec where we had to define checkstyleSpec to run :spec-tests:checkstyleMain and :spec-tests:checkstyleTest.

Now this is all done by applying the convention plugin. Any module that gets the plugin applied will get checkstyle task defined by the plugin and so when you run ./gradlew checkstyle it will run checkstyle task for all modules that define it.